### ISSUE TITLE
A question about the precision of doubles and floats in Metric

### ISSUE THREAD

**Issue by nh13** [fulcrumgenomics/fgbio#318]
@tfenne It looks like we print decimal values down to only six digits!  If we are outputting a very small double, say the error rate in duplex sequencing, then the result may be zero, when in fact it is greater than zero.  This makes me want to roll our own metric formatting utility.  We should also not forget about `Option[Double]` in the implementation.

https://github.com/fulcrumgenomics/fgbio/blob/master/src/main/scala/com/fulcrumgenomics/util/Metric.scala#L199-L205
https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/util/FormatUtil.java#L62
https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/util/FormatUtil.java#L46

**Comment by tfenne** [comment]
I do think that we should probably change it so that it formats doubles with five significant digits.  Perhaps something like:

```scala
val bigDoubleFmt = new DecimalFormat("0.######")
val smallDoubleFmt = new DecimalFormat("0.#####E0")

if (d >= 0.001) bigDoubleFmt.format(d) else smallDoubleFormat.format(d)
```

Note that this is possible to override in an individual `Metric` class as it stands.

**Comment by tfenne** [comment]
Resolved by #322
